agent/connect/ca/provider_vault.go

```diff
@@ -231,7 +244,22 @@ func (v *VaultProvider) SetIntermediate(intermediatePEM, rootPEM string) error {
 
 // ActiveIntermediate returns the current intermediate certificate.
 func (v *VaultProvider) ActiveIntermediate() (string, error) {
-	return v.getCA(v.config.IntermediatePKIPath)
+	if err := v.setupIntermediatePKIPath(); err != nil {
```

Comment:
Not sure it is the best way to make sure everything is setup in here, but I didn't find a better place. Using `setupIntermediatePKIPathDone` as a guard to make sure the setup is only done once.